pr_reviewer/llm_analyzer.py

```diff
@@ -153,122 +158,228 @@ def _build_comment_generation_prompt(self, diff: str, pr_info: dict, files: List
             prompt = base_prompt + """
-Focus on code maintainability:
-- Check code organization, structure, naming conventions
-- Look for code duplication, complex functions
-- Assess documentation, error handling
-- Identify refactoring opportunities
+Focus on code maintainability with refactoring suggestions:
+- Code organization, naming conventions → suggest improvements
+- Code duplication, complex functions → provide cleaner alternatives
+- Documentation, error handling → offer better implementations
+- Provide code suggestions to improve readability and maintainability
 
-Provide 2-5 specific maintainability-focused comments with exact file names and line numbers.
+Generate 2-5 specific maintainability-focused comments, prioritizing suggestions that clean up code.
 """
         else:  # categorized or other
             prompt = base_prompt + """
-Provide a comprehensive review covering:
-1. **Security** - Any security concerns or vulnerabilities
-2. **Performance** - Potential performance issues
-3. **Code Quality** - Maintainability, readability, best practices
-4. **Logic** - Correctness of implementation
-5. **Testing** - Missing tests or test improvements
+Provide a comprehensive review with actionable suggestions covering:
+1. **Security** - Fix security concerns with suggested secure alternatives
+2. **Performance** - Optimize slow code with efficient implementations  
+3. **Code Quality** - Improve maintainability with cleaner code suggestions
+4. **Logic** - Correct implementation issues with working alternatives
+5. **Testing** - Suggest specific test improvements or missing tests
 
-Provide 3-8 specific comments covering different aspects, with exact file names and line numbers.
+Generate 3-8 specific comments covering different aspects. Prioritize comments with code suggestions that clearly improve the codebase.
 """
         
-        return prompt + "\n\nIMPORTANT: Only comment on files that actually exist in the diff. Use the EXACT format shown above."
+        return prompt + "\n\nIMPORTANT: Only comment on files that actually exist in the diff. Use the EXACT format shown above. When providing ORIGINAL_CODE, extract it precisely from the diff - it must match exactly for GitHub suggestions to work."
     
     def _parse_analysis_to_comments(self, analysis: str, files: List[dict]) -> List[ReviewComment]:
-        """Parse LLM analysis into structured review comments"""
-        logger.debug("🔍 Parsing analysis into structured comments...")
+        """Parse LLM analysis into structured review comments with suggestion support"""
+        logger.debug("🔍 Parsing analysis into structured comments with suggestions...")
         
         comments = []
         file_names = [f['filename'] for f in files]
         
         # Split analysis into sections and look for the structured format
         lines = analysis.split('\n')
         current_comment = {}
+        capturing_original = False

```

Comment:
The parsing logic uses multiple boolean flags which makes it complex and error-prone. Consider using a state machine pattern

```suggestion
current_comment = {}
        parse_state = "NONE"  # NONE, CAPTURING_ORIGINAL, CAPTURING_SUGGESTED
        original_lines = []
        suggested_lines = []
```